serverless.yaml

```diff
@@ -778,7 +816,63 @@ resources:
                 Condition:
                   Bool:
                     'aws:SecureTransport': false
-          Queues: [!Ref DdbToEsDLQ]
+
+      RestHookQueuePolicy:
+        Type: AWS::SQS::QueuePolicy
+        Properties:
+          Queues: [!Ref RestHookQueue]
+          PolicyDocument:
+            Statement:
+              - Effect: Deny
+                Action:
+                  - SQS:*
+                Resource:
+                  - !GetAtt RestHookQueue.Arn
+                Principal: '*'
+                Condition:
+                  Bool:
+                    'aws:SecureTransport': false
+              - Effect: Allow
+                Action:
+                  - SQS:SendMessage
+                Resource:
+                  - !GetAtt RestHookQueue.Arn
+                Principal: '*'
```

Comment:
Let's specify the principal: 
```
"Principal": {
      "Service": "sns.amazonaws.com"
    },
```

https://docs.aws.amazon.com/sns/latest/dg/sns-access-policy-use-cases.html#sns-publish-messages-to-sqs-queue